python/ray/actor.py

```diff
@@ -745,11 +717,21 @@ def _deserialization_helper(self, state, ray_forking):
         worker.check_connected()
 
         if state["ray_forking"]:
-            actor_handle_id = compute_actor_handle_id(
-                ray.ObjectID(state["previous_actor_handle_id"]),
-                state["actor_forks"])
+            actor_handle_id = ray.ObjectID(state["actor_handle_id"])
         else:
-            actor_handle_id = None
+            # Right now, if the actor handle has been pickled, we create a
+            # temporary actor handle id for invocations.
+            # TODO(pcm): This still leads to a lot of actor handles being
+            # created, there should be a better way to handle pickled
+            # actor handles.
+            # TODO(swang): Accessing the worker's current task ID is not
+            # thread-safe.
+            # TODO(swang): Unpickling the same actor handle twice in the same
+            # task will break the application, and unpickling it twice in the
+            # same actor is likely a performance bug. We should consider
+            # logging a warning in these cases.
```

Comment:
If it's going to break the application then we definitely want to log an error, right? The alternative is a crash or something?